fem/fespace.hpp

```diff
@@ -1570,6 +1589,40 @@ class FiniteElementSpace
    virtual ~FiniteElementSpace();
 };
 
+/** @brief NURBS spaces can have some atypical ownership semantics. This
+    class helps to dispatch for various cases.
+ */
+struct NURBSSpace
+{
+   enum class Type { H1, Hdiv, Hcurl, };
+
+   std::unique_ptr<FiniteElementSpace> fespace;
+   std::unique_ptr<NURBSFECollection> nurbs_fec;
+   FiniteElementCollection* mesh_fec;
+   FiniteElementCollection* fec;
```

Comment:
It would help to document that `fec` is for the function space, which may differ from `mesh_fec`.